src/main/java/org/squiddev/plethora/integration/astralsorcery/MethodsAstralSorcery.java

```diff
@@ -0,0 +1,123 @@
+	// That said, players can already name a gateway, and they could add signs to the destination
+	// so that it shows in the preview (if it is working; haven't tested)
+	// If Hellfire complains, it's easy enough to delete...
+	@PlethoraMethod(
+		modId = AstralSorcery.MODID,
+		doc = "-- Get a list of all Celestial Gateways, grouped by dimension"
+	)
+	public static Map<String, ?> getGateways(@Nonnull IContext<TileCelestialGateway> context ) {
+		Map<Integer, List<GatewayCache.GatewayNode>> nodesByDimension = CelestialGatewaySystem.instance.getGatewayCache(Side.SERVER);
+		Map<String, Object> fullOut = new HashMap<>(nodesByDimension.size());
+
+		//TODO This will break badly if dimensions aren't identified by number, e.g. 1.13, NEID, JEID, etc.
+		for (Map.Entry<Integer, List<GatewayCache.GatewayNode>> entry : nodesByDimension.entrySet()) {
+			// I was going to filter this for the current node, but that will result in excessive collection manipulation
+			// if I can't convert to a stream API chain; otherwise, I would risk modifying the actual server data...
+			List<GatewayCache.GatewayNode> dimNodes = entry.getValue();
+			LuaList<Map<String, Object>> dimOut = new LuaList<>(dimNodes.size());
+			for (GatewayCache.GatewayNode node : dimNodes) {
+				Map<String, Object> inner = new HashMap<>(4);
+				inner.put("posX", node.getX());
+				inner.put("posY", node.getY());
+				inner.put("posZ", node.getZ());
+				inner.put("name", node.display);
+				dimOut.add(inner);
+			}
+
+			//TODO Determine how to get a dimension's name; `DimensionType.getById(int).getName` is a start, but
+			// it doesn't account for named dimensions, e.g. RFTools Dimensions, Mystcraft, etc. ...
+			//REFINE I recall there being a difference between `String.valueOf` and `toString` for primitives, but I
+			// don't remember the specifics...
+			fullOut.put(String.valueOf(entry.getKey()), dimOut.asMap());
+		}
+
+		return fullOut;
+	}
+
+	//Mainly useful if a player wants to design some sort of organizer to track their progress
+	@PlethoraMethod(
+		modId = AstralSorcery.MODID,
+		module = PlethoraModules.INTROSPECTION_S,
+		doc = "-- Get this player's progress in Astral Sorcery"
+	)
+	public static Map<String, ?> getAstralProgress(@Nonnull IContext<IModuleContainer> context, @FromSubtarget EntityIdentifier.Player playerId) throws LuaException {
+		EntityPlayerMP player = playerId.getPlayer();
+
+		Map<String, Object> out = new HashMap<>();
+
+		PlayerProgress progress = ResearchManager.getProgress(player);
+
+		//Refers to the constellations that you have seen on a paper
+		//REFINE This is a bit of a kludge, even extracting the lambda to a separate method;
+		// `getSeenConstellations` returns the unlocalized name(s),
+		// so we have to do a lookup if we intend to expose the meta...
+		out.put("seenConstellations", LuaList.of(progress.getSeenConstellations(), c -> getConstellationMeta(context, c)).asMap());
```

Comment:
Hrmr, will this then include null entries? Maybe worth having a `LuaList.ofNullable` as well, or doing a map + filter + map + collect here instead (probably not, could be ugly).

Alternatively, make `getConstellationMeta` consume a `Collection<String>`, and build up the list in a loop instead. Whatever you think is cleaner :).